src/platform/packages/private/kbn-esql-editor/src/esql_editor.tsx

```diff
@@ -849,8 +850,22 @@ export const ESQLEditor = memo(function ESQLEditor({
                       monaco.KeyMod.CtrlCmd | monaco.KeyCode.Slash,
                       onCommentLine
                     );
-
                     setMeasuredEditorWidth(editor.getLayoutInfo().width);
+                    if (expandToFitQueryOnMount) {
+                      const { wrappingColumn } = editor.getLayoutInfo();
+                      const totalLines = editor
```

Comment:
What exactly are you trying to compute here? If I am not mistaken monaco api returns the lines without the need to make all these computations.

Reply:
Trying to get the number of actual lines displayed when taking word-wrapping into account, e.g if there are 5 numbered lines displayed but 3 of them wrap onto new lines, we want to compute 8 lines. Looked through the API and wasn't able to find it, if you know the method I'm missing please let me know and I'll swap it out.

Reply:
Yes you can do something like that which is cleaner:

```
                    if (expandToFitQueryOnMount) {
                      const lineHeight = editor.getOption(monaco.editor.EditorOption.lineHeight);
                      const lineCount = editor.getModel()?.getLineCount() || 1;
                      const height = editor.getTopForLineNumber(lineCount + 1) + lineHeight;
                      if (height > editorHeight) setEditorHeight(height);
```